File: 974/wrnlb.py
class Solution:
    def subarraysDivByK(self, nums: list[int], k: int) -> int:
        cs: int = 0
        map: dict[int, int] = {0: 1}
        count: int = 0
        
        for v in nums:
            cs += v
            if cs % k in map:
                count += map[cs % k]
                map[cs % k] += 1
                print(cs, map[cs % k])
            else:
                map[cs % k] = 1

        return count


def main() -> None:
    nums: list[int] = [4,5,0,-2,-3,1]
    k = 5
    print(Solution().subarraysDivByK(nums, k))


if __name__ == "__main__":
    main()

File: 2028/ts/wrnlb.ts
function missingRolls(rolls: number[], mean: number, n: number): number[] {
    // calculate sum of `n` nubmers
    const m: number = rolls.length
    let sum: number = 0
    rolls.forEach((r) => {
        sum += r
    })
    const total: number = mean * (m + n)
    sum = total - sum

    // find a possible solution to consume sum
    let nMean: number = sum / n
    if (nMean > 6 || nMean < 1) {
        return []
    }
    nMean = ~~nMean
    let remainder: number = sum % n
    let res: number[] = new Array(n)
    const gap: number = 6 - nMean
    const count: number = ~~(remainder / gap)
    remainder = remainder % gap
    let i = 0
    for (; i < count; i++) {
        res[i] = 6
    }
    if (!Number.isNaN(remainder)) {
        res[i++] = nMean + remainder
    }
    for (; i < n; i++) {
        res[i] = nMean
    }
    return res
}


(() => {
    const rolls: number[] = [3,2,4,3]
    const mean: number = 4
    const n: number = 2

    const res: number[] = missingRolls(rolls, mean, n)
    console.log(res)
})()
